hooks/useInternet.ts

```diff
@@ -4,27 +4,53 @@ import { useState, useEffect } from 'react';
 export const useInternet = () => {
   const [isOnline, setIsOnline] = useState<boolean>(true);
 
-  const updateOnlineStatus = async () => {
-    try {
-      const netInfo = await NetInfo.fetch();
-      const isConnected = Boolean(netInfo.isConnected && netInfo.isInternetReachable);
-      setIsOnline(isConnected);
-      return isConnected;
-    } catch (error) {
-      console.error('Error checking network status:', error);
-      setIsOnline(false);
-      return false;
-    }
+  const updateOnlineStatus = () => {
+    return new Promise<boolean>((resolve) => {
+      NetInfo.fetch()
+        .then((netInfo) => {
+          const isConnected = Boolean(netInfo.isConnected);
+          setIsOnline(isConnected);
+          resolve(isConnected);
+        })
+        .catch((error) => {
+          console.error('Error checking network status:', error);
+          setIsOnline(false);
+          resolve(false);
+        });
+    });
   };
 
-  const checkNetwork = async () => {
-    try {
-      const isConnected = await updateOnlineStatus();
-      return isConnected;
-    } catch (error) {
-      console.error('Error in checkNetwork:', error);
-      return false;
-    }
+  const checkNetwork = () => {
+    return new Promise<boolean>((resolve) => {
+      if (isOnline) {
+        resolve(true);
+        return;
+      }
+      const timeout = setTimeout(() => {
+        resolve(false);
+      }, 500);
+
+      NetInfo.fetch()
+        .then((netInfo) => {
+          clearTimeout(timeout);
+          const isConnected = Boolean(netInfo.isConnected);
+          setIsOnline(isConnected);
+          resolve(isConnected);
+        })
+        .catch((error) => {
+          clearTimeout(timeout);
+          console.error('Error in checkNetwork:', error);
```

Comment:
We can remove this console message

Reply:
There are bunch of other console messages in the code, can you take care of those too, if they aren't added for any specific reason.